### PR TITLE
Mark Firecracker v1.4 as EOL

### PR DESCRIPTION
**Pull request by pb8o**
Firecracker v1.4 is no longer supported as of 2024-01-20.

## Changes

...

## Reason

...

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: docs/RELEASE_POLICY.md [hunk 1]
```diff
@@ -84,16 +84,16 @@ also be specifying the supported kernel versions.
...
-|    v1.3 |   2023-03-02 |       v1.3.3 |          2023-09-02 | 2023-10-09 (v1.5 released)     |
-|    v1.2 |   2022-11-30 |       v1.2.1 |          2023-05-30 | 2023-07-20 (v1.4 released)     |
-|    v1.1 |   2022-05-06 |       v1.1.4 |          2022-11-06 | 2023-03-02 (v1.3 released)     |
-|    v1.0 |   2022-01-31 |       v1.0.2 |          2022-07-31 | 2022-11-30 (v1.2 released)     |
-|   v0.25 |   2021-03-13 |      v0.25.2 |          2021-09-13 | 2022-03-13 (end of 1y support) |
+| Release | Release Date | Latest Patch | Min. end of support | Official end of Support         |
+|--------:|-------------:|-------------:|--------------------:|:--------------------------------|
+|    v1.6 |   2023-12-20 |       v1.6.0 |          2024-06-20 | Supported                       |
+|    v1.5 |   2023-10-09 |       v1.5.1 |          2024-04-09 | Supported                       |
+|    v1.4 |   2023-07-20 |       v1.4.1 |          2024-01-20 | 2023-10-09 (end of 6mo support) |
```
[the remark this answers is not in the document]
Comment (pb8o) [reply docs/RELEASE_POLICY.md:91]: :facepalm:

[the remark this answers is not in the document]
Comment (pb8o) [reply docs/RELEASE_POLICY.md]: Fixed

**Review by roypat — APPROVED** [review]
:ship:

**Review by xmarcalx — APPROVED** [review]
LGTM, thanks for the change

**Pull request merged**